devicetypes/vision-nameseca/zp3111jp-4-in-1-motion.src/zp3111jp-4-in-1-motion.groovy

```diff
@@ -0,0 +1,925 @@
+	if (state.refreshAll || !device.currentValue("motion")) {
+		sendMotionEvents(cmd.value)
+	}
+	return []
+}
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicSet cmd) {
+	logTrace "BasicSet: $cmd"
+	sendMotionEvents(cmd.value)
+	return []
+}
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	logTrace "NotificationReport: $cmd"
+	def result = []
+	if (cmd.notificationType == 7) {
+		if (cmd.eventParameter[0] == 3 || cmd.event == 3) {
+			result += handleTamperEvent(cmd.v1AlarmLevel)
+		}
+	}
+	return result
+}
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+	logTrace "SensorMultilevelReport: ${cmd}"
+
+	def eventMaps = []
+	switch (cmd.sensorType) {
+		case tempSensorType:
+			def unit = tempUnits.find { it.value == cmd.scale }?.unit
+			def temp = cmd.scaledSensorValue
+			eventMaps += createTempEventMaps(temp, false)
+			break
+		case humiditySensorType:
+			eventMaps += createHumidityEventMaps(cmd.scaledSensorValue, false)
+			break
+		case lightSensorType:
+			eventMaps += createLightEventMaps(cmd.scaledSensorValue, false)
+			break
+	}
+
+	eventMaps += createStatusEventMaps(eventMaps, false)
+
+	def result = []
+	eventMaps?.each {
+		logTrace "Creating Event: ${it}"
+		it.descriptionText = getDisplayedDescriptionText(it)
+		result << createEvent(it)
+	}
+	return result
+}
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	logDebug "Unhandled Command: $cmd"
+	return []
+}
+private createCheckIntervalEvent(seconds) {
+	// Set the Health Check interval so that it reports offline 5 minutes after it's missed 2 checkins.
+	def val = ((seconds * 2) + (5 * 60))
+
+	def eventMap = createEventMap("checkInterval", val, "", false)
+
+	eventMap.data = [protocol: "zwave", hubHardwareId: device.hub.hardwareID]
+
+	return createEvent(eventMap)
+}
+private sendLastCheckinEvent() {
+	if (!isDuplicateCommand(state.lastCheckinTime, 60000)) {
+		state.lastCheckinTime = new Date().time
+
+		sendEvent(createEventMap("lastCheckin", convertToLocalTimeString(new Date()), "", false))
+	}
+}
+private canReportBattery() {
+	def reportEveryMS = (batteryReportingIntervalSettingSeconds * 1000)
+
+	return (!state.lastBatteryReport || ((new Date().time) - state.lastBatteryReport > reportEveryMS))
+}
+private sendUpdatingEvent() {
+	if (getAttrValue("pendingChanges") != -1) {
+		sendEvent(createEventMap("pendingChanges", -1, "", false))
+	}
+}
+private sendMotionEvents(val) {
+	def motionVal = (val == 0xFF ? "active" : "inactive")
+
+	def eventMaps = []
+	eventMaps += createEventMaps("motion", motionVal, "", true, false)
+	eventMaps += createStatusEventMaps(eventMaps, false)
+
+	eventMaps?.each {
+		it.descriptionText = getDisplayedDescriptionText(it)
+		sendEvent(it)
+	}
+}
+private handleTamperEvent(val)
+{
+	def result = []
+	def tamperVal
+
+	if (val == 0xFF)
+	{
+		tamperVal = "detected"
+	}
+	else if (val == 0)
+	{
+//		if (autoClearTamperSetting)
```

Comment:
Remove the commented code

Reply:
I have removed the commented code.
thank you.